devicetypes/smartthings/zwave-window-shade.src/zwave-window-shade.groovy

```diff
@@ -15,252 +15,262 @@ import groovy.json.JsonOutput
-    response(refresh())
+	sendEvent(name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+	sendEvent(name: "supportedWindowShadeCommands", value: JsonOutput.toJson(["open", "close", "pause"]), displayed: false)
+	response(refresh())
 }
 
 def updated() {
-    if (device.latestValue("checkInterval") != checkInterval) {
-        sendEvent(name: "checkInterval", value: checkInterval, displayed: false)
-    }
-    if (!device.latestState("battery")) {
-        response(zwave.batteryV1.batteryGet())
-    }
+	if (device.latestValue("checkInterval") != checkInterval) {
+		sendEvent(name: "checkInterval", value: checkInterval, displayed: false)
+	}
+	if (!device.latestState("battery")) {
+		response(zwave.batteryV1.batteryGet())
+	}
 }
 
 def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
-    handleLevelReport(cmd)
+	handleLevelReport(cmd)
 }
 
 def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd) {
-    handleLevelReport(cmd)
+	handleLevelReport(cmd)
 }
 
 def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelReport cmd) {
-    handleLevelReport(cmd)
+	handleLevelReport(cmd)
 }
 
 private handleLevelReport(physicalgraph.zwave.Command cmd) {
-    def descriptionText = null
-    def shadeValue = null
-
-    def level = cmd.value as Integer
-    if (level >= 99) {
-        level = 100
-        shadeValue = "open"
-    } else if (level <= 0) {
-        level = 0  // unlike dimmer switches, the level isn't saved when closed
-        shadeValue = "closed"
-    } else {
-        shadeValue = "partially open"
-        descriptionText = "${device.displayName} shade is ${level}% open"
-    }
-    checkLevelReport(level)
-    def levelEvent = createEvent(name: "level", value: level, unit: "%", displayed: false)
-    def stateEvent = createEvent(name: "windowShade", value: shadeValue, descriptionText: descriptionText, isStateChange: levelEvent.isStateChange)
-
-    def result = [stateEvent, levelEvent]
-    if (!state.lastbatt || now() - state.lastbatt > 24 * 60 * 60 * 1000) {
-        log.debug "requesting battery"
-        state.lastbatt = (now() - 23 * 60 * 60 * 1000) // don't queue up multiple battery reqs in a row
-        result << response(["delay 15000", zwave.batteryV1.batteryGet().format()])
-    }
-    result
+	def descriptionText = null
+	def shadeValue = null
+
+	def level = cmd.value as Integer
+	if (level >= 99) {
+		level = 100
+		shadeValue = "open"
+	} else if (level <= 0) {
+		level = 0  // unlike dimmer switches, the level isn't saved when closed
+		shadeValue = "closed"
+	} else {
+		shadeValue = "partially open"
+		descriptionText = "${device.displayName} shade is ${level}% open"
+	}
+	checkLevelReport(level)
+
+	def levelEvent = createEvent(name: "level", value: level, unit: "%", displayed: false)
```

Comment:
@dkirker
Since level and shadeLevel keep the same value - I'm wondering why levelEvent is still needed. Is it because of existing automations (so not to break anything) ? Are there any plans for migrating such automations ?
What about the UI (metadata) ? Are we going to have two sliders to set level value?

Reply:
@KKlimczukS `Switch Level` is going away, but it can't be pulled out wholesale. The UI metadata will be updated accordingly.

Reply:
Step 1 is to add `Window Shade Level`.
Step 2 is to use `Window Shade Level` for the device details control and both as rule options, removing `Switch Level` as the device details control.
Step 3 is to remove `Switch Level` from use with rules and migrate all existing to `Window Shade Level`.

It might be easier to just have step 2 and step 3 combined as step 2 to reduce the risk of duplicate rules.